core_impl/src/main/java/io/opencensus/implcore/tags/TaggerImpl.java

```diff
@@ -31,19 +33,27 @@
   // TODO(sebright): Consider treating an unknown TagContext as empty.  That would allow us to
   // remove TagContext.unsafeGetIterator().
 
+  private final AtomicReference<TaggingState> state;
```

Comment:
private volatile TaggingState state.

You don't need getAndSet or lazySet.

Reply:
I want you just to replace your current line with my proposed line:
s/private final AtomicReference<TaggingState> state;/private volatile TaggingState state;

Reply:
I don't understand how that change would work.  How would the TagsComponent update the state?

Reply:
Oh what you do is you share an atomic reference object between multiple classes. I would like to propose changes:
1) Create a class for the CurrentState (or Eanbled).
2) Share that object if you don't want to have it as a global state (using static functions).

What do you think? That is much cleared that you share that state that is owned by the TraceComponent.

Reply:
Done